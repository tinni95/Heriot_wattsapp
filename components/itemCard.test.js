import React from "react";
import ItemCard from "./itemCard";
import renderer from "react-test-renderer";

props = {
    price: 20,
    name: "item",
};

describe("itemCard", () => {
    let rendered;
    beforeEach(() => {
        rendered = renderer.create(<ItemCard {...props} />);
    });

    it("renders", () => {
        expect(rendered).toBeTruthy();
    });

    it("matches snapshot", () => {
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
