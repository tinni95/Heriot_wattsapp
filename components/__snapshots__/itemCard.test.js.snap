// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemCard matches snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginTop": 20,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F4F4F4",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "row",
        "height": 100,
        "margin": 8,
        "width": 320,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 6,
          "flexDirection": "column",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        item
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginTop": 15,
          }
        }
      >
        Â£
        20
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#348653",
          "borderBottomRightRadius": 10,
          "borderTopRightRadius": 10,
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        +
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 22,
        }
      }
    >
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={false}
        visible={false}
      >
        <View
          style={
            Object {
              "marginTop": 22,
            }
          }
        >
          <View>
            <Text>
              Hello World!
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <Text
                style={null}
              >
                Hide Modal
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
